src/main/java/io/supertokens/cronjobs/bulkimport/ProcessBulkImportUsers.java

```diff
@@ -113,230 +123,19 @@ public int getInitialWaitTimeSeconds() {
         return 0;
     }
 
-    private synchronized Storage getBulkImportProxyStorage(TenantIdentifier tenantIdentifier)
-            throws InvalidConfigException, IOException, TenantOrAppNotFoundException, DbInitException {
-        String userPoolId = StorageLayer.getStorage(tenantIdentifier, main).getUserPoolId();
-        if (userPoolToStorageMap.containsKey(userPoolId)) {
-            return userPoolToStorageMap.get(userPoolId);
-        }
-
-        TenantConfig[] allTenants = Multitenancy.getAllTenants(main);
-
-        Map<ResourceDistributor.KeyClass, JsonObject> normalisedConfigs = Config.getNormalisedConfigsForAllTenants(
-                allTenants,
-                Config.getBaseConfigAsJsonObject(main));
-
-        for (ResourceDistributor.KeyClass key : normalisedConfigs.keySet()) {
-            if (key.getTenantIdentifier().equals(tenantIdentifier)) {
-                SQLStorage bulkImportProxyStorage = (SQLStorage) StorageLayer.getNewBulkImportProxyStorageInstance(main,
-                        normalisedConfigs.get(key), tenantIdentifier, true);
-
-                userPoolToStorageMap.put(userPoolId, bulkImportProxyStorage);
-                bulkImportProxyStorage.initStorage(false, new ArrayList<>());
-                return bulkImportProxyStorage;
-            }
-        }
-        throw new TenantOrAppNotFoundException(tenantIdentifier);
-    }
-
-    private Storage[] getAllProxyStoragesForApp(Main main, AppIdentifier appIdentifier)
-            throws StorageTransactionLogicException {
-
-        try {
-            List<Storage> allProxyStorages = new ArrayList<>();
-            TenantConfig[] tenantConfigs = Multitenancy.getAllTenantsForApp(appIdentifier, main);
-            for (TenantConfig tenantConfig : tenantConfigs) {
-                allProxyStorages.add(getBulkImportProxyStorage(tenantConfig.tenantIdentifier));
-            }
-            return allProxyStorages.toArray(new Storage[0]);
-        } catch (TenantOrAppNotFoundException e) {
-            throw new StorageTransactionLogicException(new Exception("E043: " + e.getMessage()));
-        } catch (InvalidConfigException e) {
-            throw new StorageTransactionLogicException(new InvalidConfigException("E044: " + e.getMessage()));
-        } catch (DbInitException e) {
-            throw new StorageTransactionLogicException(new DbInitException("E045: " + e.getMessage()));
-        } catch (IOException e) {
-            throw new StorageTransactionLogicException(new IOException("E046: " + e.getMessage()));
+    private List<List<BulkImportUser>> makeChunksOf(List<BulkImportUser> users, int numberOfChunks) {
```

Comment:
If there were million users to be imported, are all of them loaded to memory first?